sds/src/scanner/mod.rs

```diff
@@ -261,16 +249,19 @@ impl Scanner {
 
         // Measure detection time
         let start = std::time::Instant::now();
-        self.scoped_ruleset.visit_string_rule_combinations(
-            event,
-            ScannerContentVisitor {
-                scanner: self,
-                caches,
-                rule_matches: &mut rule_matches_list,
-                blocked_rules: &blocked_rules_idx,
-                excluded_matches: &mut excluded_matches,
-            },
-        );
+        access_regex_caches(|regex_caches| {
```

Comment:
FYI I will very soon work on a scan cache (for ML Rules) so this will be renamed pretty quickly :D

Reply:
This only works for regexes, so I'm not sure why it needs to be renamed. If ML Rules wants to cache a regex, this will still work :)

Reply:
no but we want to cache other things